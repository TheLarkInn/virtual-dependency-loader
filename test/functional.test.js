// import fs from 'fs';
// import Promise from 'bluebird';
import {
  runWebpackExampleInMemory,
  // fs as sharedFs
} from '../test/test-utils';

// const readdir = Promise.promisify(fs.readdir, { context: sharedFs });
// const readFile = Promise.promisify(fs.readFile, { context: sharedFs });
// const fsReaddir = Promise.promisify(fs.readdir, { context: sharedFs });
// const fsReadFile = Promise.promisify(fs.readFile, { context: sharedFs });
// const fsStat = Promise.promisify(fs.stat, { context: sharedFs });
// const fsExists = Promise.promisify(fs.exists, { context: sharedFs });

test('should run with no errors or warnings', async () => {
  const buildStats = await runWebpackExampleInMemory('simple');
  const { errors, warnings } = buildStats;

  expect([...errors, ...warnings].length).toBe(0);
});

test('should append transformations to JavaScript module', async () => {
  const buildStats = await runWebpackExampleInMemory('simple');
  const { modules } = buildStats;
  const moduleToTest = modules[3].source()._source._value; //eslint-disable-line
  const loadedString = 'export default "virtual"';

  expect(moduleToTest).toEqual(expect.stringContaining(loadedString));
  expect(moduleToTest).toMatchSnapshot();
});
