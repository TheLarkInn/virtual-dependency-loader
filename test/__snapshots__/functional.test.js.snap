// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should append transformations to JavaScript module 1`] = `
"
  /**
   * Resource Location: /Users/wecomm/seanlarkin/code/example-webpack-loader/src/index.js
   * Loaders chainded to module: [{\\"path\\":\\"/Users/wecomm/seanlarkin/code/example-webpack-loader/src/index.js\\",\\"query\\":\\"??ref--0-0\\",\\"options\\":{},\\"ident\\":\\"ref--0-0\\",\\"data\\":null,\\"pitchExecuted\\":true,\\"normalExecuted\\":true,\\"request\\":\\"/Users/wecomm/seanlarkin/code/example-webpack-loader/src/index.js??ref--0-0\\"}]
   * Loader API Version: 2
   * Is this in \\"webpack mode\\": true
   * This is the users request for the module: /Users/wecomm/seanlarkin/code/example-webpack-loader/src/index.js??ref--0-0!/Users/wecomm/seanlarkin/code/example-webpack-loader/src/index.js
   */

  /**
   * Original Source From Loader
   */

  export default function loader(source) {
  const { loaders, resource, request, version, webpack } = this;

  const newSource = \`
  /**
   * Resource Location: \${resource}
   * Loaders chainded to module: \${JSON.stringify(loaders)}
   * Loader API Version: \${version}
   * Is this in \\"webpack mode\\": \${webpack}
   * This is the users request for the module: \${request}
   */

  /**
   * Original Source From Loader
   */

  \${source}\`;

  return newSource;
}
"
`;
